test/e2e/dra/test-driver/app/gomega.go

```diff
@@ -30,3 +32,13 @@ var BeRegistered = gcustom.MakeMatcher(func(actualCalls []GRPCCall) (bool, error
 	}
 	return false, nil
 }).WithMessage("contain successful NotifyRegistrationStatus call")
+
+// NodePrepareResouceCalled checks that NodePrepareResource API has been called
+var NodePrepareResourceCalled = gcustom.MakeMatcher(func(actualCalls []GRPCCall) (bool, error) {
+	for _, call := range actualCalls {
```

Comment:
I don't think this is the case. Otherwise we'd need to do this for every loop in the code.

Reply:
Alright, I am misunderstanding then. @rsc you recently reported multiple issues related to what you called the loopvar sharing bug. Just for my understanding when do we fall into that situation ? Does it apply to every for range loop or is there specific criteria ?

Reply:
It's not needed here because the loop variable is only used inside the loop body.

The loopvar problem occurs when the value is used after the loop is complete (referencing it in a goroutine, in a parallel test (which is ultimately a goroutine), storing a pointer to it, etc.).

Reply:
Thanks Ed and Patrick, it's crystal clear now :)